src/Validation/Requires.cs

```diff
@@ -457,6 +457,79 @@ public static void That([DoesNotReturnIf(false)] bool condition, string? paramet
             }
         }
 
+        /// <summary>
+        /// Throws an exception if the specified parameter's value is not null
+        /// <em>and</em> has an element which does not match the given predicate.
+        /// </summary>
+        /// <typeparam name="T">The type of the elements in the sequence.</typeparam>
+        /// <param name="values">The value of the argument.</param>
+        /// <param name="predicate">The predicate used to test the elements.</param>
+        /// <param name="parameterName">The name of the parameter to include in any thrown exception.</param>
+        /// <param name="message">A message to be used in the resulting exception.</param>
+        /// <exception cref="ArgumentException">Thrown if the tested condition is false.</exception>
+        /// <exception cref="ArgumentNullException">Thrown if <paramref name="predicate"/> is <see langword="null"/>.</exception>
+        [DebuggerStepThrough]
+        public static void ValidElements<T>([ValidatedNotNull] IEnumerable<T> values, Predicate<T> predicate, string parameterName, string message)
```

Comment:
Your null ref annotations suggest that `parameterName` and `message` must not be null. But we do not validate that. Does the `ArgumentException` constructor that we throw require that? It looks from [here](https://source.dot.net/#System.Private.CoreLib/ArgumentException.cs,56) that it allows null values for these arguments. So presumably we should too.

Reply:
I missed those. I'll update it.